src/pull-request.ts

```diff
@@ -1,23 +1,35 @@
-import * as sourcegraph from 'sourcegraph'
+import { app, commands, configuration, ExtensionContext, NotificationType, search } from 'sourcegraph'
+import { runCodemodQuery } from './codemod'
 import { issuePullRequestWithToken } from './pull-request-core'
+import { bundlePullRequests } from './user-input-sequence'
 
-/**
- * The activate function is called when one of the extensions `activateEvents`
- * conditions in package.json are satisfied.
- */
-export function activate(ctx: sourcegraph.ExtensionContext): void {
-    // const gitHubAccessToken = sourcegraph.configuration
-    //    .get()
-    //    .get('github.pr.accessToken')
-    // if (!gitHubAccessToken) {
-    //    console.log('Could not read github.pr.accessToken from configuration')
-    //    return;
-    // }
-
-    const gitHubAccessToken = 'YOUR_GH_TOKEN'
+export function activate(ctx: ExtensionContext): void {
+    // store the query string in configuration after a search. It's used to execute a GQL codemod command when we're ready to PR.
+    ctx.subscriptions.add(
+        search.registerQueryTransformer({
+            transformQuery: async query => {
+                await commands.executeCommand('updateConfiguration', 'pr.codemodQuery', query)
+                return query
+            },
+        })
+    )
     ctx.subscriptions.add(
-        sourcegraph.commands.registerCommand('pr.issuePullRequest', (pullRequest, changesetCommit) =>
-            issuePullRequestWithToken({ pullRequest, changesetCommit }, gitHubAccessToken)
-        )
+        commands.registerCommand('pr.issuePullRequest', async (pullRequest, changesetCommit) => {
+            const query = configuration.get().value['pr.codemodQuery']
+            if (query) {
+                if (!app.activeWindow) {
+                    throw new Error('No active window')
+                }
+                app.activeWindow.showNotification(`🔥 Preparing your PR, hang on hot second`, NotificationType.Info)
```

Comment:
Sounds like the progress API would be appropriate here

Reply:
I tried at first but it didn't behave well with the alerts (notification would not disappear before the alert box popped up). Probably something I'm for controlling that interaction. I'll revisit and give more detail on the issues I ran into then.

Reply:
With "alert/input box", do you mean the user input prompt? That is synchronous in JS, so no UI can be updated while a prompt is open as it is implemented rn. But why is that a problem?

Do you mean you were unable to complete the progress before a prompt? You can complete the progress notification with `reporter.complete()`